android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -358,6 +367,67 @@ public void captureScreenshot(Promise promise) {
         promise.resolve(screenshotsArray);
     }
 
+    public void fetchViewHierarchy(Promise promise) {
+        final @Nullable Activity activity = getCurrentActivity();
+        final @Nullable View decorView = getCurrentDecorViewForViewHierarchy(activity);
+
+        if (decorView == null) {
+            promise.resolve(null);
+            return;
+        }
+
+        final @NotNull ViewHierarchy viewHierarchy = ViewHierarchyEventProcessor.snapshotViewHierarchy(decorView);
+
+        byte[] bytes;
+        try {
+            try (final @NotNull ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                 final @NotNull BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(stream, UTF_8));
+                 final @NotNull JsonObjectWriter jsonWriter = new JsonObjectWriter(writer, 100)) {
+
+                viewHierarchy.serialize(jsonWriter, logger);
+                jsonWriter.flush();
+                bytes = stream.toByteArray();
+            }
+        } catch (Throwable t) {
+            logger.log(SentryLevel.ERROR, "Could not serialize ViewHierarchy", t);
+            promise.resolve(null);
+            return;
+        }
```

Comment:
H: I'd instead prefer that `ViewHierarchyEventProcessor` exposes a function that returns the array of bytes directly since the Android SDK already knows everything needed, this function only needs the `Activity`, less code to maintain, less duplication, and less memory usage since you don't need to create your own serializer, iOS is already like that.

Reply:
@krystofwoldrich the change has to live in the Android SDK, so @markushi and @romtsn maybe can assist you with the changes or do it for you.